kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerManagementIntegrationTest.java

```diff
@@ -408,6 +416,137 @@ public void testGetNotExistingContainer() {
+
+        assertNotNull(containerList);
+        assertEquals(1, containerList.size());
+
+        ContainerSpec containerResponseEntity = containerList.iterator().next();
+        assertNotNull(containerResponseEntity);
+        assertEquals(CONTAINER_ID, containerResponseEntity.getId());
+        assertEquals(CONTAINER_NAME, containerResponseEntity.getContainerName());
+        assertEquals(releaseId, containerResponseEntity.getReleasedId());
+        assertEquals(KieContainerStatus.STOPPED, containerResponseEntity.getStatus());
+    }
+
+    @Test
+    public void testEmptyListContainers() {
+        try {
+            Collection<ContainerSpec> emptyList = controllerClient.listContainerSpec(kieServerInfo.getServerId());
+            fail("Should throw exception about kie server instance not existing.");
+        } catch (UnexpectedResponseCodeException e) {
+            assertEquals(404, e.getResponseCode());
+        }
+
+        // Create kie server instance connection in controller.
+        createServerTemplate();
+
+        Collection<ContainerSpec> emptyList = controllerClient.listContainerSpec(kieServerInfo.getServerId());
+        assertNullOrEmpty("Active containers found!", emptyList);
+    }
+
+    @Test
+    public void testUpdateContainerConfig() {
+        // Create kie server instance connection in controller.
+        ServerTemplate serverTemplate = createServerTemplate();
+
+        Map<Capability, ContainerConfig> containerConfigMap = new HashMap();
+
+        ProcessConfig processConfig = new ProcessConfig("PER_PROCESS_INSTANCE", "kieBase", "kieSession", "MERGE_COLLECTION");
+        containerConfigMap.put(Capability.PROCESS, processConfig);
+
+        RuleConfig ruleConfig = new RuleConfig(500l, KieScannerStatus.SCANNING);
+        containerConfigMap.put(Capability.RULE, ruleConfig);
+
+        ContainerSpec containerSpec = new ContainerSpec(CONTAINER_ID, CONTAINER_NAME, serverTemplate, releaseId, KieContainerStatus.STOPPED, containerConfigMap);
+        controllerClient.saveContainerSpec(serverTemplate.getId(), containerSpec);
+
+        ContainerSpec containerResponseEntity = controllerClient.getContainerInfo(kieServerInfo.getServerId(), CONTAINER_ID);
+        assertNotNull(containerResponseEntity);
+        assertEquals(CONTAINER_ID, containerResponseEntity.getId());
+        assertEquals(releaseId, containerResponseEntity.getReleasedId());
+        assertEquals(KieContainerStatus.STOPPED, containerResponseEntity.getStatus());
+
+        // Check process config and rule config
+        checkContainerConfig(kieServerInfo.getServerId(), CONTAINER_ID, processConfig, ruleConfig);
+
+        processConfig = new ProcessConfig("SINGLETON", "defaultKieBase", "defaultKieSession", "OVERRIDE_ALL");
+        controllerClient.updateContainerConfig(kieServerInfo.getServerId(), CONTAINER_ID, Capability.PROCESS, processConfig);
```

Comment:
UpdateContainerConfig returns response with code 400. Expect that update successful change values.